fem/bilininteg.hpp

```diff
@@ -2127,6 +2129,39 @@ class GradientIntegrator : public BilinearFormIntegrator
     can be a scalar or a matrix coefficient. */
 class DiffusionIntegrator: public BilinearFormIntegrator
 {
+public:
+
+   using KernelType = void(*)(const int, const bool, const Array<real_t>&,
```

Comment:
It would be great to have this capability for a wide variety of integrators, but of course it will require doing this kind of work to each of them one by one.  I think it would be useful to write up a website HowTo on taking existing integrators and explaining what you need to add to enable this.